datafusion/functions-aggregate-common/src/aggregate/groups_accumulator/bool_op.rs

```diff
@@ -77,7 +82,9 @@ where
 
         if self.values.len() < total_num_groups {
             let new_groups = total_num_groups - self.values.len();
-            self.values.append_n(new_groups, Default::default());
+            // Fill with the identity element, so that when the first non-null value is encountered,
+            // it will combine with the identity and the result will be the first non-null value itself.
+            self.values.append_n(new_groups, self.identity);
```

Comment:
Nice -- this is similar to the default value for the PrimitiveOp accumulator too